internal/acctest/const/consts_or_quote_gen.go

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+// Code generated by internal/generate/acctestconsts/main.go; DO NOT EDIT.
+
+package acctest
+
+import (
+	"fmt"
+)
+
+// ConstOrQuote returns the constant name for the given attribute if it exists.
+// Otherwise, it returns the attribute quoted. This is intended for use in
+// generated code and templates.
+func ConstOrQuote(constant string) string {
```

Comment:
Is this just to save time when compiling and linking? It only takes a long time for the first one and then the rest are cached so they run at normal speed. Diff between generation with/without is only a minute or so. I'm not a fan of proliferating packages unless there's a compelling reason.

Reply:
During a `make gen` the first one takes a lot of time, but each additional service package takes an extra 10-15 seconds. So as we add generated tests to more services, it will add a lot of time

Reply:
I've been thinking about this, and because it's generator-related (but not a generator itself) we could name this `internal/acctest/generate` or something similar:

* being a sub-package of `acctest` makes it clear that it is directly associated with `acctest`
* calling it `generate` (or similar) indicates that it's intended for use from generators and shouldn't be used elsewhere
* it's a separate package: `acctest` is used **in** acctests, `acctest/generate` is used by generators to **create** acctests

Reply:
Aah I see, it's just for generated test code.